src/components/Modal/ModalHeader.test.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import { fireEvent, render } from '@testing-library/react'
+import { LocaleContext } from '../../i18n'
+import ptBr from '../../i18n/locales/pt-BR'
+import { ModalHeader, ModalHeaderProps } from './ModalHeader'
+
+const defaultProps: ModalHeaderProps = {
+  title: 'Title',
+  backgroundColor: 'red',
+  hasCloseIcon: true,
+  onCloseButtonClick: jest.fn(),
+}
+
+it('should render correctly', () => {
+  const { container } = render(<ModalHeader {...defaultProps} />)
+
+  expect(container).toMatchSnapshot()
+})
+
+it('should render with string title', () => {
+  const { getByText } = render(<ModalHeader {...defaultProps} />)
+
+  expect(getByText('Title')).toBeInTheDocument()
+})
+
+it('should render with JSX element as title', () => {
+  const jsxTitle = <div data-testid='custom-jsx-element'>Custom JSX Element</div>
+  const { getByTestId } = render(<ModalHeader {...defaultProps} title={jsxTitle} />)
+
+  expect(getByTestId('custom-jsx-element')).toBeInTheDocument()
+})
+
+it('should apply backgroundColor prop correctly', () => {
+  const { container } = render(<ModalHeader {...defaultProps} />)
```

Comment:
```suggestion
  const { container } = render(<ModalHeader {...defaultProps} backgroudColor='red' />)
```
Always pass the relevant properties in the tests.